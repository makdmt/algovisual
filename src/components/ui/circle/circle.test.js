import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';
import renderer from 'react-test-renderer'

describe('Тесты компонента Circle', () => {
    it('Рендеринг Circle без содержимого соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle с буквой внутри соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle letter="M" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle с меткой head соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle head="head" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle с дополнительным Circle в head соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle head={<Circle />} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle с меткой tail соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle tail="tail" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle с дополнительным Circle в tail соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle tail={<Circle />} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle с меткой index соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle index="8" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle с пропсом isSmall={true} соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle isSmall={true} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle в состоянии default соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Default} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle в состоянии changing соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Changing} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендеринг Circle в состоянии modified соответствует снепшоту', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Modified} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})

