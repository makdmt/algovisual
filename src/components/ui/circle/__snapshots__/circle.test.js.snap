// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты компонента Circle Рендеринг Circle без содержимого соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle   default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  />
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle в состоянии changing соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle   changing"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  />
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle в состоянии default соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle   default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  />
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle в состоянии modified соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle   modified"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  />
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle с буквой внутри соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle   default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      M
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  />
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle с дополнительным Circle в head соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head-circle"
      />
      <div
        className="circle   default"
        data-testid="main-circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="tail-index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail-circle"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  />
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle с дополнительным Circle в tail соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle   default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head-circle"
      />
      <div
        className="circle   default"
        data-testid="main-circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="tail-index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail-circle"
      />
    </div>
  </div>
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle с меткой head соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head-circle"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  />
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle с меткой index соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle   default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  >
    8
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail-circle"
  />
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle с меткой tail соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle   default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail-circle"
  >
    tail
  </div>
</div>
`;

exports[`Тесты компонента Circle Рендеринг Circle с пропсом isSmall={true} соответствует снепшоту 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-circle"
  />
  <div
    className="circle  small default"
    data-testid="main-circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="tail-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-circle"
  />
</div>
`;
